### ISSUE TITLE
CI: still seeing "fatal: unsafe repository"

### ISSUE THREAD

**Issue by cbm755** [cbm755/octsympy#1161]
```
fatal: unsafe repository ('/home/jovyan/octsympy' is owned by someone else)
To add an exception for this directory, call:
	git config --global --add safe.directory /home/jovyan/octsympy
```
Maybe we need to `git config` as well (or instead) of `sudo git config`...   At least I don't think that "global" thing means "all users".  More like "all repos"

**Comment by alexvong1995** [comment]
The current situation is kind of awkward and complicated. Since https://github.com/actions/checkout/issues/760 is now closed (it was still opened when we introduced the workarounds if I recalled correctly), one might think the issue should have been resolved.

Unfortunately, testing seems to suggest otherwise. The situation can be summarized using the following tables:

| Previous results | bist                                              | doctests                                               | *-pythonic      |
|------------------|---------------------------------------------------|--------------------------------------------------------|-----------------|
| w/o workarounds  | failed to build with "fatal: unsafe repository"   | failed to build with "fatal: unsafe repository"        | (no such tests) |
| w/ workarounds   | build successfully w/o "fatal: unsafe repository" | build successfully but show "fatal: unsafe repository" | (no such tests) |


| Current results | bist, doctests                                         | bist-pythonic, doctests-pythonic                  |
|-----------------|--------------------------------------------------------|---------------------------------------------------|
| w/o workarounds | build successfully but show "fatal: unsafe repository" | failed to build with "fatal: unsafe repository"   |
| w/ workarounds  | build successfully but show "fatal: unsafe repository" | build successfully w/o "fatal: unsafe repository" |

**Comment by cbm755** [comment]
Was that before or after c6edc2ad311813485a01ca33c5fd5ea5bafc7302?  I thought that cleaned up some of the messages.

**Comment by alexvong1995** [comment]
Yeah, there was before c6edc2ad311813485a01ca33c5fd5ea5bafc7302, I was looking at the main branch only. I'll update the tables.

**Comment by cbm755** [comment]
Let's merge that PR first.
